camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/policy/GroupsChangePolicyOsgiTest.java

```diff
@@ -0,0 +1,472 @@
+        Asserts.eventually(() -> member.getAttribute(Sensors.newStringSensor(MEMBER_SENSOR_TO_CREATE)), input -> input != null && input.length() == SENSOR_VALUE);
+    }
+
+    /**
+     * Tests {@link GroupsChangePolicy#INITIALIZERS} of the member while joining the {@link DynamicGroup},
+     * {@link CompositeEffector} in particular since it has a peculiar way to resolve effectors to call together.
+     */
+    @Test
+    public void testAddInitializers_CompositeEffector() throws Exception {
+
+        // Config names.
+        final String COMPOSITE_EFFECTOR_NAME = "composite-effector";
+
+        // Blueprint to test.
+        String yaml = Joiner.on("\n").join(
+                "services:",
+                "- type: " + DynamicGroup.class.getName(),
+                "  brooklyn.policies:",
+                "  - type: " + GroupsChangePolicy.class.getName(),
+                "    brooklyn.config:",
+                "      group: $brooklyn:self()",
+                "      member.initializers:",
+                "      - type: " + EffectorForThisTest.class.getName(),
+                "      - type: " + CompositeEffector.class.getName(),
+                "        brooklyn.config:",
+                "          name: " + COMPOSITE_EFFECTOR_NAME, // <--- THIS IS THE EFFECTOR TO CALL IN THIS TEST
+                "          effectors:",
+                "            - " + EffectorForThisTest.NAME,
+                "- type: " + TestEntity.class.getName());
+
+        final Entity app = createStartWaitAndLogApplication(yaml);
+        Asserts.assertNotNull(app);
+
+        final DynamicGroup dynamicGroup = (DynamicGroup) Iterables.getFirst(app.getChildren(), null);
+        Asserts.assertNotNull(dynamicGroup);
+
+        final TestEntity member = (TestEntity) Iterables.getLast(app.getChildren());
+        Asserts.assertNotNull(member);
+
+        // Verify that member has joined.
+        dynamicGroup.setEntityFilter(EntityPredicates.idEqualTo(member.getId()));
+        Asserts.assertEquals(dynamicGroup.getMembers().size(), 1);
+
+        // Wait for sensor that is expected to be added by EffectorForThisTest, which is expected to be called by CompositeEffector configured above.
+        Asserts.eventually(() -> {
+
+            // Call the composite effector as member got it from the GroupsChangePolicy.
+            Effector<?> helloEffector = member.getEffector(COMPOSITE_EFFECTOR_NAME);
+            if (!Objects.isNull(helloEffector)) {
+                invokeEffector(member, helloEffector); // <-- THE EFFECTOR TO CALL IN THIS TEST
+            }
+
+            return member.getAttribute(Sensors.newStringSensor(EffectorForThisTest.SENSOR));
+        }, EffectorForThisTest.SENSOR_VALUE::equals);
+    }
+
+    /**
+     * Tests {@link GroupsChangePolicy#POLICIES} of the member while joining the {@link DynamicGroup},
+     * {@link InvokeEffectorOnSensorChange} in particular.
+     */
+    @Test
+    public void testAddPolicies() throws Exception {
```

Comment:
This is test it has been rewritten for YAML test fixture and a different policy.